Improve experience when going to Komponenter

wkillerud opened the pull request:
This PR introduces a new build system for the styleguide CSS. It
lets Webpack handle it by extracting the final CSS with
`ExtractTextWebpackPlugin` if the build is a production build. We use
`NODE_ENV` to determine if we're building PROD or not, and use
`cross-env` for compatability with other platforms.

Combined with `HtmlWebpackPlugin` which comes with React Styleguidist
this setup creates a separate CSS while which is linked in `head`. This
should fix the issue of the flash of unstyled content reported in #48.

The PR also brings changes to the Handlebars templates to improve
the loading experience while we're fetching the styleguide JavaScript
bundle, which can take a while.

Finally I bring an improvement to the DX when working with Less/CSS locally
by fixing hot reloading/live changes of styles :tada:

wkillerud committed:
fix: Fix live reloading of CSS

We were depending on rebuilds of CSS to get styles. When developing
locally this gets tedious. With this commit the styleguide now uses
`ffe-all.less` instead of `dist/ffe.css` so live reloading works.

wkillerud committed:
fix: Remove redundant script

Running `styleguidist` by itself does nothing. Use the `:server` or
`:build` scripts instead.

wkillerud committed:
fix: Improve experience when going to Komponenter

This commit introduces a new build system for the styleguide CSS. It
lets Webpack handle it by extracting the final CSS with
`ExtractTextWebpackPlugin` if the build is a production build. We use
`NODE_ENV` to determine if we're building PROD or not, and use
`cross-env` for compatability with other platforms.

Combined with `HtmlWebpackPlugin` which comes with React Styleguidist
this setup creates a separate CSS while which is linked in `head`. This
should fix the issue of the flash of unstyled content reported in #48.

The commit also brings changes to the Handlebars templates to improve
the loading experience while we're fetching the styleguide JavaScript
bundle, which can take a while.

selbekk reviewed (dismissed):
MERGE IT RIGHT AWAY

kwltrs commented on package.json:
```diff
@@ -42,11 +40,13 @@
         "babel-preset-stage-3": "^6.24.1",
         "case": "^1.5.4",
         "coveralls": "^3.0.0",
+        "cross-env": "^5.1.3",
```
> [...] and use `cross-env` for compatability with other platforms.

Other platforms, that´s windows, or?

wkillerud replied:
Windows, yes. Figured I could use a cross platform util since I think this is the first use of environment variables in the project.

wkillerud commented:
Resolved a merge conflict with the macOS fix by @antidecaf :dancing_men:

wkillerud committed:
fix: Remove NODE_ENV from test scripts

Jest sets this to `test` automatically, so no need to clutter our
`package.json`.

selbekk reviewed (dismissed):
great improve!
